### DISCUSSION TITLE
Question about fetching images

### DISCUSSION THREAD

**Discussion by captain-cronch** [joalla/discogs_client#131] [Support] [answered]
I’ve been trying to fetch release images, but every time I get a URL that I think leads to a release image, the file is always 249.3 kb and is unable to be open. I have opened one of the urls and there is no image appearing. I am authenticated through my user key. 

In addition, it appears release images all generate urls indicating a size of maximum 600x600.

Any help or sample code would be greatly appreciated. 
Thanks.

**Comment by AnssiAhola** [comment]
Hey @captain-cronch 

Can you provide us with some example releases that this happened with?

I tried the following and got an valid link to an image.
```python
print(client.release(18424870).images[0]["uri"])
# https://i.discogs.com/rbgsspwMTX95LTy4GJEGCl4gFouCyOHIySrTHM0vFWI/rs:fit/g:sm/q:90/h:600/w:600/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTE4NDI0/ODcwLTE2MTkxODc2/OTktNDEyOS5qcGVn.jpeg
```
![image](https://github.com/joalla/discogs_client/assets/43060692/46b7272c-ca7c-43ff-b295-24e53f49aeee)

**Reply by captain-cronch** [reply]
I’ll try this release in my code and see how it returns. Is it possible it’s an authorization issue? I haven’t had issue pulling information from releases in my collection otherwise and have an authorization key in my code. 

I’ll check on it and upload my code when I’m back home.

**Reply by AnssiAhola** [reply]
It seems like it's possible that the authorization is the issue here.

[https://www.discogs.com/developers/#page:authentication,header:authentication-discogs-auth-flow](https://www.discogs.com/developers/#page:authentication,header:authentication-discogs-auth-flow)

Theres a table at the bottom of that section that indicates that returning image urls requires valid authorization.
You'd think that the link would be just empty, not invalid 🤔

**Comment by captain-cronch** [comment]
Specificially, this is the code I am currently working with:

```
import discogs_client
import os
import requests

discogs = discogs_client.Client('attributetest', user_token='<usertoken>')

user = discogs.user('<myuser>')
releases = user.collection_folders[0].releases

with open('artist.txt', 'r') as f:
    artist = f.readline().strip().lower()
with open('song.txt', 'r') as f:
    song = f.readline().strip().lower()

for r in releases:
    if r.release.artists[0].name.lower() == artist:
        for track in r.release.tracklist:
            if track.title.lower() == song:
                directory = os.path.join('art', r.release.artists[0].name)
                if not os.path.exists(directory):
                    os.makedirs(directory)

                response = requests.get(r.release.images[0]["uri"])
                with open(os.path.join(directory, f"{r.release.title}.jpg"), 'wb') as f:
                    f.write(response.content)

                print(f"{r.release.artists[0].name} - {r.release.title}")
                print(r.release.images[0]["uri"])
                break
```

print(r.release.images[0]["uri"]) returns valid urls that I am able to view on my PC. I am attempting to save the files on a raspberry pi.

The most recent release I tested was 11155125. From what I can tell, personal access token should be sufficient. I tested release 18424870 and the same holds true, I can return the url, but have no luck saving the image.

Edit: additional update, I tried oAuth instead of a personal access token and I still wasn't able to save files. I think I forgot to mention it, but when I have tried print(response) in the past, I receive a 403 error and I'm still not entirely sure why

**Reply by captain-cronch** [reply, marked as answer]
After perusing the discogs API forum some more, it looks like there's a chance that cloudflare has anti-scraping measures that prevent me from downloading the file without taking some counter measures first. I'll be looking into that, but if anyone has any solutions, I'd love to hear them.

**Reply by JOJ0** [reply]
Please take a minute and format your code to show properly using markdown (triple bacticks wrapping) Thanks a lot
